File: servers/jax-rs/src/main/java/com/wordnik/api/LeaderboardApi.java
package com.wordnik.api;

import com.wordnik.swagger.annotations.*;

import com.wordnik.client.model.Leaderboard;
import java.util.List;
import com.wordnik.api.NotFoundException;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/leaderboard.json")
@Api(value = "/leaderboard", description = "the leaderboard API")
@Produces({"application/json"})
public class LeaderboardApi {
  @GET
  @Path("/")
  @ApiOperation(value = "Get current leaderboard.", notes = "", responseClass = "List<leaderboard>")
  @ApiErrors(value = { })
     
  public Response leaderboard_getOrderBook(
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  }

File: servers/scalatra/src/main/scala/com/wordnik/client/api/LeaderboardApi.scala
package com.wordnik.client.api

import com.wordnik.client.model.Leaderboard

import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class LeaderboardApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "LeaderboardApi"
  override protected val applicationName: Option[String] = Some("Leaderboard")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }
  

  val leaderboard.getOperation = (apiOperation[List[Leaderboard]]("leaderboard.get")
      summary "Get current leaderboard."
      parameters(queryParam[String]("method").description("").optional.defaultValue(notional))
  )

  get("/leaderboard",operation(leaderboard.getOperation)) {
    
    
    
        
      
      val method = params.getAs[String]("method")
            

    
    println("method: " + method)
  
  }

}